<!DOCTYPE html>
<html>
<head>
	<meta charset='utf-8'>
	<title>Schedule</title>
	<meta name="description" content="Schedule">
	<meta name="author" content="Wei Cui">
	<meta name="generator" content="slidify" />
	
	<!-- LOAD STYLE SHEETS -->
	<link rel="stylesheet" href="libraries/frameworks/html5slides/default/styles.css">
	<link rel="stylesheet" href="libraries/highlighters/prettify/css/tomorrow.css" />  <link rel="stylesheet" href = "assets/css/ribbons.css">
  
</head>
<body style='display: none'>
	<section class='slides layout-regular template-regular'>
     <article class="" id="slide-1" style="background:;">
  <h3>20150313 job</h3>
  <ol>
<li>Watch the video of MultiGeMS</li>
<li>Check EM algorithm of R code</li>
<li>Check EM algorithm of C++ code</li>
<li>Select all suspicious sites from GiaB and Upenn datasets (unpooled)</li>
<li>Find some region from true SNPs according to properties (such as coverage, quality, etc.)</li>
<li>Find some special region according to properties</li>
</ol>

</article>

<article class="" id="slide-2" style="background:;">
  <h3>Check EM algorithm (R)</h3>
  <ul>
<li>Use all the default value from GeMS-3gen-submax.R</li>
<li>Process data from file &quot;samples&quot;</li>
<li>Run multiEM function</li>
</ul>

</article>

<article class="" id="slide-3" style="background:;">
  <h3>Code (Without paste of default values)</h3>
  <pre><code>require(&quot;minqa&quot;)
require(&quot;Biostrings&quot;)
data &lt;- read.table(file=&quot;samples&quot;, header=FALSE, sep=&quot;\t&quot;)
data &lt;- cbind(data, data[,1])
data &lt;- data[, 4:8]
data &lt;- t(data)
source(&quot;em.R&quot;)
multiEM(&quot;C&quot;, data, c(1),1)
</code></pre>

</article>

<article class="" id="slide-4" style="background:;">
  <h3>Check EM algorithm (C++)</h3>
  <ul>
<li>Modify code to add debug function</li>
<li>Modify code to add test function</li>
<li>Run without parameter</li>
</ul>

</article>

<article class="" id="slide-5" style="background:;">
  <h3>Create site file interest.txt</h3>
  <ul>
<li>Use upennmed.R (Comment pool part)</li>
<li>Use gzip -d *.vcf.gz if necessary</li>
<li>Use gzip -d *.out.gz if necessary</li>
<li>Use cut -f 1-3 -d . multigemslist if necessary</li>
<li>Use regions.rda</li>
<li>Use intersect() setdiff()</li>
</ul>

</article>

<article class="" id="slide-6" style="background:;">
  <h3>Select suspicious sites</h3>
  <pre><code>zgrep -f ~/interest.txt *.pile.gz &gt; ~/sites.txt
sort -n -k 2 sites.txt &gt; sites.sorted.txt
</code></pre>

<p>completeness check</p>

<pre><code>cat sites.sorted.txt | cut -f 2 | uniq | wc -l
cat interest.txt | wc -l
cat sites.sorted.txt | cut -c 3-7 | sort | uniq | wc -l
</code></pre>

</article>

<article class="" id="slide-7" style="background:;">
  <h3>Histgram</h3>
  <ul>
<li>The whole dataset</li>
<li>Every Chromosome</li>
</ul>

</article>

<article class="" id="slide-8" style="background:;">
  <h3>GiaB dataset</h3>
  <pre><code class="r">library(ggplot2)
load(&quot;GiaB_dataset.rda&quot;)
p &lt;- ggplot(data=GiaB_dataset, aes(x=Chr, fill=factor(Snp)))
p + geom_bar(position=&#39;stack&#39;)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-1-1.png" title="plot of chunk unnamed-chunk-1" alt="plot of chunk unnamed-chunk-1" style="display: block; margin: auto;" /></p>

</article>

<article class="" id="slide-9" style="background:;">
  <h3>GiaB dataset</h3>
  <pre><code class="r">library(ggplot2)
load(&quot;GiaB_dataset.rda&quot;)
p &lt;- ggplot(data=GiaB_dataset, aes(x=Chr, fill=factor(Snp)))
p + geom_bar(position=&#39;fill&#39;)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-2-1.png" title="plot of chunk unnamed-chunk-2" alt="plot of chunk unnamed-chunk-2" style="display: block; margin: auto;" /></p>

</article>

<article class="" id="slide-10" style="background:;">
  <h3>GiaB dataset (Code)</h3>
  <pre><code class="r">library(ggplot2); 
load(&quot;GiaB_dataset.rda&quot;)
p &lt;- ggplot(data=GiaB_dataset[GiaB_dataset$Chr==&quot;chr19&quot;, ], aes(Site))
p + geom_histogram(binwidth=500000, position=&#39;identity&#39;, 
alpha=0.5, aes(y=..density.., fill=factor(Snp))) + 
stat_density(geom=&#39;line&#39;, position = &#39;identity&#39;, aes(colour = factor(Snp))) +
facet_grid(Snp~.)
</code></pre>

</article>

<article class="" id="slide-11" style="background:;">
  <h3>GiaB dataset (Chr18)</h3>
  <p><img src="assets/fig/unnamed-chunk-4-1.png" title="plot of chunk unnamed-chunk-4" alt="plot of chunk unnamed-chunk-4" style="display: block; margin: auto;" /></p>

</article>

<article class="" id="slide-12" style="background:;">
  <h3>GiaB dataset (Chr19)</h3>
  <p><img src="assets/fig/unnamed-chunk-5-1.png" title="plot of chunk unnamed-chunk-5" alt="plot of chunk unnamed-chunk-5" style="display: block; margin: auto;" /></p>

</article>

<article class="" id="slide-13" style="background:;">
  <h3>GiaB dataset (Chr20)</h3>
  <p><img src="assets/fig/unnamed-chunk-6-1.png" title="plot of chunk unnamed-chunk-6" alt="plot of chunk unnamed-chunk-6" style="display: block; margin: auto;" /></p>

</article>

<article class="" id="slide-14" style="background:;">
  <h3>GiaB dataset (Chr21)</h3>
  <p><img src="assets/fig/unnamed-chunk-7-1.png" title="plot of chunk unnamed-chunk-7" alt="plot of chunk unnamed-chunk-7" style="display: block; margin: auto;" /></p>

</article>

<article class="" id="slide-15" style="background:;">
  <h3>GiaB dataset (Chr22)</h3>
  <p><img src="assets/fig/unnamed-chunk-8-1.png" title="plot of chunk unnamed-chunk-8" alt="plot of chunk unnamed-chunk-8" style="display: block; margin: auto;" /></p>

</article>

<article class="" id="slide-16" style="background:;">
  <h3>Upenn dataset</h3>
  <pre><code class="r">library(ggplot2)
load(&quot;upenn_dataset.rda&quot;)
p &lt;- ggplot(data=upenn_dataset, aes(x=Chr, fill=factor(Snp)))
p + geom_bar(position=&#39;stack&#39;)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-9-1.png" title="plot of chunk unnamed-chunk-9" alt="plot of chunk unnamed-chunk-9" style="display: block; margin: auto;" /></p>

</article>

<article class="" id="slide-17" style="background:;">
  <h3>Upenn dataset</h3>
  <pre><code class="r">library(ggplot2)
load(&quot;upenn_dataset.rda&quot;)
p &lt;- ggplot(data=upenn_dataset, aes(x=Chr, fill=factor(Snp)))
p + geom_bar(position=&#39;fill&#39;)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-10-1.png" title="plot of chunk unnamed-chunk-10" alt="plot of chunk unnamed-chunk-10" style="display: block; margin: auto;" /></p>

</article>

<article class="" id="slide-18" style="background:;">
  <h3>Upenn dataset (Code)</h3>
  <pre><code class="r">library(ggplot2); 
load(&quot;upenn_dataset.rda&quot;)
p &lt;- ggplot(data=upenn_dataset[upenn_dataset$Chr==&quot;chr19&quot;, ], aes(Site))
p + geom_histogram(binwidth=500000, position=&#39;identity&#39;, 
alpha=0.5, aes(y=..density.., fill=factor(Snp))) + 
stat_density(geom=&#39;line&#39;, position = &#39;identity&#39;, aes(colour = factor(Snp))) +
facet_grid(Snp~.)
</code></pre>

</article>

<article class="" id="slide-19" style="background:;">
  <h3>Upenn dataset (Chr19)</h3>
  <p><img src="assets/fig/unnamed-chunk-12-1.png" title="plot of chunk unnamed-chunk-12" alt="plot of chunk unnamed-chunk-12" style="display: block; margin: auto;" /></p>

</article>

<article class="" id="slide-20" style="background:;">
  <h3>Upenn dataset (Chr20)</h3>
  <p><img src="assets/fig/unnamed-chunk-13-1.png" title="plot of chunk unnamed-chunk-13" alt="plot of chunk unnamed-chunk-13" style="display: block; margin: auto;" /></p>

</article>

<article class="" id="slide-21" style="background:;">
  <h3>Upenn dataset (Chr21)</h3>
  <p><img src="assets/fig/unnamed-chunk-14-1.png" title="plot of chunk unnamed-chunk-14" alt="plot of chunk unnamed-chunk-14" style="display: block; margin: auto;" /></p>

</article>

<article class="" id="slide-22" style="background:;">
  <h3>Upenn dataset (Chr22)</h3>
  <p><img src="assets/fig/unnamed-chunk-15-1.png" title="plot of chunk unnamed-chunk-15" alt="plot of chunk unnamed-chunk-15" style="display: block; margin: auto;" /></p>

</article>

<article class="" id="slide-23" style="background:;">
  <h3>20150313 learning</h3>
  <ol>
<li>R basic manipulation</li>
<li>slidify</li>
<li>ggplot2</li>
<li>googleVis</li>
</ol>

</article>

<article class="" id="slide-24" style="background:;">
  <h3>STUDYING</h3>
  <ol>
<li>Least Angle Regression</li>
<li>Robust subspace clustering</li>
<li>A Survey of Clustering Data Mining Techniques</li>
<li>Sparse Recovery via Differential Inclusions</li>
<li>ROP: Matrix Recovery via Rank-one Projections</li>
<li><font color='red'>ParticleCall</font></li>
</ol>

</article>

<article class="" id="slide-25" style="background:;">
  <h3>Something about RStudio</h3>
  <ol>
<li>The default device of RStudio maybe has some problem when plot</li>
<li>Try to use <code>options(device=&quot;X11&quot;)</code> or other devices</li>
<li>Library Cairo is required if use alpha parameter which make the graphic pretty</li>
<li>If R can not find Cairo, <font color='red'>do not</font> use alpha parameter</li>
</ol>

</article>

  </section>
</body>
  <!-- LOAD JAVASCRIPTS  -->
	<script src='libraries/frameworks/html5slides/default/slides.js'></script>
	<!-- Google Prettify -->
	<script src="http://cdnjs.cloudflare.com/ajax/libs/prettify/188.0.0/prettify.js"></script>
	<script src='libraries/highlighters/prettify/js/lang-r.js'></script>
	<script>
	  var pres = document.getElementsByTagName("pre");
	  for (var i=0; i < pres.length; ++i) {
	    pres[i].className = "prettyprint linenums";
	  }
	  prettyPrint();
	</script>
	<!-- End Google Prettify --> 
	</html>
